lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package regexp provides regular expression related functions.
+package regexp // import "go.starlark.net/lib/regexp"
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module regexp is a Starlark module of regular expression related functions.
+// The module defines the following functions:
+//
+//     compile(pattern) - Compile a regular expression pattern into a regular expression object, which
+//                        can be used for matching using its matches(), find(), find_all and other methods.
+//
+//     re.find(src) - Returns a string holding the text of the leftmost match in the given string of the regular expression re.
+//                    If there is no match, the return value is an empty string, but it will also be empty if the
+//                    regular expression successfully matches an empty string.
+//
+//     re.find_all(src, max) - Returns a tuple of all successive matches of the regular expression re. An empty tuple indicates
+//                             no match. If max > 0, at most max strings will be returned. If max == 0,
+//                             an empty tuple will be returned. If max < 0, all strings will be returned.
+//                             The parameter max is optional, by default no limit will be applied.
+//
+//     re.find_submatches(src) - Returns a tuple of strings holding the text of the leftmost match of the regular expression re
+//                               in the given string and the matches, if any, of its subexpressions. An empty tuple indicates
+//                               no match.
+//
+//     re.matches(src) - Indicates whether the given string contains any match of the regular expression re.
+//
+//     re.replace_all(src , repl) - Returns a copy of the given string, replacing matches of the regular expression re
+//                                  with the replacement string repl. Inside repl, $ signs are interpreted as in Expand,
```

Comment:
What is Expand?

What's with the $ treatment? This is not a feature of RE2 (and other RE packages use \1 notation for them).